### PR TITLE
libpmi: enable flux to bootstrap with cray libpmi2.so

### PR DESCRIPTION
**Pull request by garlick**
This works through the problems identified in #5040.  I've been able to test this on rzvernal, where the cray PMI libraries were added to LD_LIBRARY_PATH and seems ok.

```
$ srun -N2 -n4 src/cmd/flux start flux getattr broker.mapping
[[0,2,2,1]]
$ src/cmd/flux start flux getattr broker.mapping
(snip big warning from libhwloc)
[[0,1,1,1]]
```

### CODE REVIEW

**Comment by garlick** [comment]
Probably should hold off here - I'm seeing the occasional segfault when launching under slurm and want to run that down.

**Comment by garlick** [comment]
This is based on top of #5053 now, so that one should go in first.

I've been unable to reproduce the segfault I saw earlier so I'll say (tentatively) that this is tentatively ready for a review.

File: src/common/libpmi/upmi_libpmi2.c
Comment (grondo) [inline src/common/libpmi/upmi_libpmi2.c:139]: :joy:

**Comment by garlick** [comment]
Rebased on master after #5053 was merged.

File: src/broker/boot_pmi.c
Comment (grondo) [inline src/broker/boot_pmi.c:91]: Just to clarify, the broker still does need to fall back to PMI_process_mapping here?

File: t/t3002-pmi.t
Comment (grondo) [inline t/t3002-pmi.t:209]: I'm unclear on why the test should _not_ ensure that both libpmi2 and libpmi plugins refuse to boot from Flux's own library. Maybe the commit message just needs to explain the why of the problem statement. (Apologies if it should be obvious)

**Review by grondo — APPROVED** [review]
Nice!

Just a couple questions inline. I didn't find anything major so approving.

File: src/broker/boot_pmi.c
Comment (garlick) [reply src/broker/boot_pmi.c:91]: Yes, the current code which does
```c
        char *s;
        if (upmi_get (upmi, "flux.taskmap", -1, &s, NULL) == 0
            || upmi_get (upmi, "PMI_process_mapping", -1, &s, NULL) == 0) {
            val = pmi_mapping_to_taskmap (s);
            free (s);
        }
```
is fine I think!  In any case, the issue was closed so the reference probably should be dropped.

File: t/t3002-pmi.t
Comment (garlick) [reply t/t3002-pmi.t:209]: It could check both, but it says its checking libpmi and I add another check for libpmi2 later on, so it was more just an effort to organize tests by plugin and not have the behavior of one plugin affect a test that claims to be checking the other. I'll update the commit message to try to be more clear.

**Comment by garlick** [comment]
Repushed with that commit message changed to
```
    testsuite: narrow libpmi noflux test
    
    Problem: a test that claims to check the libpmi plugin's rejection
    of Flux's libpmi.so is now checking libpmi2 also
    
    Set FLUX_PMI_CLIENT_METHODS to skip over libpmi2 so a failure will
    implicate the correct plugin.
```
and I'll go ahead and set MWP.

File: t/t3002-pmi.t
Comment (grondo) [reply t/t3002-pmi.t:209]: Ah! That makes sense (it should have been obvious, sorry!)

**Pull request merged**
